### ISSUE TITLE
Starting marytts-server fails

### ISSUE THREAD

**Issue by entenbein** [marytts/marytts#369]
Hi folks,

this happens with the version from git from Monday:

I start the server like 
```
sh marytts-server
```

This is the output in the console:
```
MARY server 5.2-SNAPSHOT starting as a HTTP server...Exception in thread "main" marytts.exceptions.MaryConfigurationException: Cannot instantiate object from 'marytts.language.de.JPhonemiser': 1
	at marytts.util.MaryRuntimeUtils.instantiateObject(MaryRuntimeUtils.java:135)
	at marytts.modules.ModuleRegistry.instantiateModule(ModuleRegistry.java:87)
	at marytts.server.Mary.startModules(Mary.java:130)
	at marytts.server.Mary.startup(Mary.java:291)
	at marytts.server.Mary.startup(Mary.java:201)
	at marytts.server.Mary.main(Mary.java:491)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
	at marytts.modules.JPhonemiser.readLexicon(JPhonemiser.java:369)
	at marytts.modules.JPhonemiser.<init>(JPhonemiser.java:109)
	at marytts.modules.JPhonemiser.<init>(JPhonemiser.java:87)
	at marytts.language.de.JPhonemiser.<init>(JPhonemiser.java:73)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at java.lang.Class.newInstance(Class.java:383)
	at marytts.util.MaryRuntimeUtils.instantiateObject(MaryRuntimeUtils.java:131)
	... 5 more
Exception in thread "Thread-1" java.lang.IllegalStateException: MARY system is not running
	at marytts.server.Mary.shutdown(Mary.java:363)
	at marytts.server.Mary$3.run(Mary.java:284)
```

This is from the server.log:

2015-08-05 13:22:00,983 [main] INFO  marytts.main Mary starting up...
2015-08-05 13:22:00,983 [main] INFO  marytts.main Specification version 5.2-SNAPSHOT
2015-08-05 13:22:00,983 [main] INFO  marytts.main Implementation version unknown
2015-08-05 13:22:00,983 [main] INFO  marytts.main Running on a Java 1.7.0_85 implementation by Oracle Corporation, on a Linux platform (amd64, 2.6.32-504.23.4.el6.x86_64)
2015-08-05 13:22:01,217 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.en.JTokeniser'
2015-08-05 13:22:01,221 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.en.Preprocess'
2015-08-05 13:22:01,369 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(en_US.)'
2015-08-05 13:22:01,457 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(en_GB.)'
2015-08-05 13:22:01,515 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.en.Prosody'
2015-08-05 13:22:01,518 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.SimplePhoneme2AP(en_US)'
2015-08-05 13:22:01,519 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.en.PronunciationModel'
2015-08-05 13:22:01,520 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.OpenNLPPosTagger(en,en.pos)'
2015-08-05 13:22:01,520 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(sv.)'
2015-08-05 13:22:01,538 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.MinimalisticPosTagger(sv,sv.)'
2015-08-05 13:22:01,539 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.it.JTokeniser'
2015-08-05 13:22:01,539 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.it.Preprocess'
2015-08-05 13:22:01,540 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(it.)'
2015-08-05 13:22:01,564 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.OpenNLPPosTagger(it,it.pos)'
2015-08-05 13:22:01,564 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.it.Prosody'
2015-08-05 13:22:01,565 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.it.Postlex'
2015-08-05 13:22:01,565 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.Synthesis'
2015-08-05 13:22:01,566 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.TextToMaryXML'
2015-08-05 13:22:01,567 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.SableParser'
2015-08-05 13:22:01,567 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.SSMLParser'
2015-08-05 13:22:01,568 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.APMLParser'
2015-08-05 13:22:01,568 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.EmotionmlParser'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JTokeniser'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.DummyTokens2Words'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.ProsodyGeneric'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.PronunciationModel'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.TargetFeatureLister'
2015-08-05 13:22:01,570 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.HalfPhoneTargetFeatureLister'
2015-08-05 13:22:01,570 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.AcousticModeller'
2015-08-05 13:22:01,571 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.RealisedAcoustparamsExtractor'
2015-08-05 13:22:01,571 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.RealisedDurationsExtractor'
2015-08-05 13:22:01,572 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.HTSEngine'
2015-08-05 13:22:01,573 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.PraatTextGridGenerator'
2015-08-05 13:22:01,574 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.te.JTokeniser'
2015-08-05 13:22:01,574 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.DummyTokens2Words'
2015-08-05 13:22:01,574 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.te.JPhonemiser(te.)'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.ProsodyGeneric'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.MinimalisticPosTagger(te,te)'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.ru.Tokenizer'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.DummyTokens2Words'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(ru.)'
2015-08-05 13:22:01,751 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.ProsodyGeneric'
2015-08-05 13:22:01,751 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.MinimalisticPosTagger(ru,ru)'
2015-08-05 13:22:01,751 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.de.JTokeniser'
2015-08-05 13:22:01,752 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.de.Preprocess'
2015-08-05 13:22:01,752 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.OpenNLPPosTagger(de,de.pos)'
2015-08-05 13:22:01,753 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.de.JPhonemiser'

With an older version of marytts in a different location, everything works fine...

Who can help?

Thanks!

**Comment by psibre** [comment]
The stacktrace is not specific enough, but it might be that your userdict is malformed.

**Comment by entenbein** [comment]
Hi,

this only seems to occur if I place the user-dictionary in *$marybase/target/marytts-5.2-SNAPSHOT/user-dictionaries/*. Still with the user-dictionary (only?) in *$marybase/user-dictionaries/* for some of my words within that dictionary only g2p rules are applied (see #368).

Is it enough to place to user-dictionary in *$marybase/user-dictionaries/* ? Do I need to put it elsewhere as well?

Thanks!

**Comment by psibre** [comment]
Maven copies the `user-dictionaries` from the `project.basedir` (i.e., your git repository root) to the `project.build.directory` (i.e., `target`) in the `package` phase (because it's explicitly included in the [runtime descriptor](../blob/014200417da6b2540dbe599073e9d40daa803a07/marytts-assembly/assembly-runtime/src/main/descriptors/runtime.xml#L14)). So your custom entries should go into the former location, not under `target`, and depending on how you run MaryTTS, you may need to rerun `mvn package` (or manually copy them if you prefer).

**Comment by entenbein** [comment]
Alright, that worked! Thanks

**Comment by psibre** [comment]
Glad to hear it's solved.

But I'm keeping this issue open until we've implemented a test that guards against this sort of thing happening in the future.

**Comment by psibre** [comment]
This is patched in 1869adbff0e9a4e1948ee58d12ac9ba63783a835; unit tests would be nice, but too much overhead atm.
